src/main/java/com/solvd/app/dao/PersonDAO.java

```diff
@@ -0,0 +1,182 @@
+                person.setEmail(resultSet.getString("email"));
+                person.setGender(resultSet.getString("gender"));
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to get person by name and phone number: " + e.getMessage());
+        } finally {
+            connectionPool.releaseConnection(connection);
+        }
+
+        return person;
+    }
+
+    public void updateEntity(Person person) {
+        Connection connection = connectionPool.getConnection();
+
+        try {
+            String sql = "UPDATE people SET name = ?, address = ?, phone_number = ?, email = ?, gender = ? WHERE person_id = ?";
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setString(1, person.getName());
+            preparedStatement.setString(2, person.getAddress());
+            preparedStatement.setInt(3, person.getPhoneNumber());
+            preparedStatement.setString(4, person.getEmail());
+            preparedStatement.setString(5, person.getGender());
+            preparedStatement.setInt(6, person.getPersonID());
+
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows == 0) {
+                LOGGER.error("Updating failed, no rows affected.");
+            } else {
+                LOGGER.info("Updated the person with ID number " + person.getPersonID());
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to update person: " + e.getMessage());
+        } finally {
+            connectionPool.releaseConnection(connection);
+        }
+    }
+
+    public void deleteEntityByID(int id) {
+        Connection connection = connectionPool.getConnection();
+
+        try {
+            String sql = "DELETE FROM people WHERE person_id = ?";
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, id);
+
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows == 0) {
+                LOGGER.error("Deleting failed, no rows affected.");
+            } else {
+                LOGGER.info("Deleted the person with ID number " + id);
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to delete person: " + e.getMessage());
+        } finally {
+            connectionPool.releaseConnection(connection);
+        }
+    }
+
+    public List<Person> getAll() {
+        Connection connection = connectionPool.getConnection();
+        List<Person> personList = new ArrayList<>();
+
+        try {
+            String sql = "SELECT * FROM people";
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            while (resultSet.next()) {
+                Person person = new Person();
```

Comment:
Why not use the custom constructor you've implemented in models/Person?

Reply:
I created these constructors for testing a part of my code and forgot to remove them. I have just deleted them. Thank you for pointing that out.